libsolidity/formal/BMC.cpp

```diff
@@ -184,7 +185,39 @@ bool BMC::visit(IfStatement const& _node)
 		);
 	m_context.popSolver();
 
-	SMTEncoder::visit(_node);
+	_node.condition().accept(*this);
+	auto conditionExpr = expr(_node.condition());
+	// visit true branch
+	auto indicesBeforeBranch = copyVariableIndices();
+	pushPathCondition(conditionExpr);
+	_node.trueStatement().accept(*this);
+	auto trueEndPathCondition = currentPathConditions();
+	popPathCondition();
+	auto indicesEndTrue = copyVariableIndices();
+	resetVariableIndices(indicesBeforeBranch);
+	auto touchedVars = touchedVariables(_node.trueStatement());
+
+	// visit false branch
+	decltype(indicesEndTrue) indicesEndFalse;
+	pushPathCondition(!conditionExpr);
+	auto falseEndPathCondition = currentPathConditions();
+	if (_node.falseStatement())
+	{
+		indicesBeforeBranch = copyVariableIndices();
+		_node.falseStatement()->accept(*this);
+		falseEndPathCondition = currentPathConditions();
+		indicesEndFalse = copyVariableIndices();
+		resetVariableIndices(indicesBeforeBranch);
+		touchedVars += touchedVariables(*_node.falseStatement());
```

Comment:
Ugh.. github throw my comment way here. This block is similar to the `true` one above. I think you can just modify `visitBranch` (which is not being used right now) to do that instead.

Reply:
Actually, `visitBranch` is still used for `Conditional` and in boolean operations.
I was thinking if those can effect path condition, but right now I think they don't, so that should be fine. What do you think?

Reply:
Ah true. Hmm I don't know for sure that there would be no effect, since conditionals and boolean operators can call functions which are inlined and their branches might do something to this? Can you add some tests to check that?

Reply:
I think the function calls inside conditionals and booleans should be fine, it is not going to have an effect on the path condition when that function call returns.
What would you like to test in this case?

Reply:
So you're saying it won't have an effect because we push and pop the currentPathConditions around an inlined function call?

Reply:
Yes, I believe that is what I am saying :)